starknet_py/net/deployer/contract_deployment.py

```diff
@@ -0,0 +1,89 @@
+from typing import Optional, List, Union
+
+from starkware.starknet.definitions.fields import ContractAddressSalt
+from starkware.starknet.public.abi import get_selector_from_name
+
+from starknet_py.net.client_models import Hash, InvokeFunction, Call
+from starknet_py.utils.contructor_args_translator import translate_constructor_args
+from starknet_py.utils.data_transformer.universal_deployer_serializer import (
+    universal_deployer_serializer,
+    deploy_contract_abi,
+    deploy_contract_event_abi,
+)
+
+
+class ContractDeployment:
+    """
+    ContractDeployment used to prepare and send deploy invoke transactions
+    """
+
+    def __init__(
+        self, deployer: "Deployer", class_hash: Hash, abi: Optional[List] = None
```

Comment:
Can we do something to avoid string types? Perhaps nesting this class in Deployer? As `Deployer.Deployment`?

Reply:
Will try